lib/_stream_writable.js

```diff
@@ -107,6 +112,13 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // when `state._end()` was called
+  this.flushed = false;
+
+  // NOTE: added here to not pollute the prototype of the WritableStream and to
+  // avoid conflicts with user-land methods
+  this._end = options._end || _end;
```

Comment:
Nah, it won't work. `_end` is implemented outside of `_stream_writable` and I can't ask it to pass `stream` and `state`.